### ISSUE TITLE
Update Wire so that it won't fall back to non generate code if MethodReaderInterceptorReturns is specified 

### ISSUE THREAD

**Issue by RobAustin** [OpenHFT/Chronicle-Wire#217]
Please implement this todo

![image](https://user-images.githubusercontent.com/6506056/95738936-f1160a00-0c81-11eb-8e84-1ded0c3f2da6.png)

**Comment by peter-lawrey** [comment]
Ideally this should check is a sub-interface has been used, something like

```java
interface GeneratingMethodReaderInterceptorReturns extends MethodReaderInterceptorReturns {
    String generatorId(); // to use in the generated class name
    String codeBeforeCall(Method m, String objectName, String[] argumentNames);
    String codeAfterCall(Method m, String objectName, String[] argumentNames);
}
```

**Comment by glukos** [comment]
@peter-lawrey Thanks for the idea! A few questions:
1. How do you think, is it crucial to support regular `MethodReaderInterceptorReturns` in generated method reader as well? It may be less efficient than using GeneratingMethodReaderInterceptorReturns, of course, but it will allow existing users to avoid falling back to vanilla version.
2. Classic `MethodReaderInterceptorReturns` is more flexible, e.g. it allows to don't perform actual invoke at all (by just not calling Invocation#invoke). Shall we support it in codeBeforeCall contract? For example, we call allow `return;` statements in returned code.

**Comment by peter-lawrey** [comment]
Until we can retire the fallback, we will still need to support it.

Adding code can avoid the call by injecting something like

.Before
```java
if (calls) {
```

.After
```java
}
```

**Comment by glukos** [comment]
See GeneratingMethodReaderInterceptorReturns (contract is described in javadoc) and MethodReaderInterceptorReturnsTest (some examples)

**Comment by hft-team-city** [comment]
Released in [Chronicle-Wire-2.20.82](https://github.com/OpenHFT/Chronicle-Wire/releases/chronicle-wire-2.20.82), [BOM-2.20.112](https://github.com/OpenHFT/OpenHFT/releases/chronicle-bom-2.20.112)

**Comment by hft-team-city** [comment]
Released in [Chronicle-Wire-2.20.101](https://github.com/OpenHFT/Chronicle-Wire/releases/chronicle-wire-2.20.101), [BOM-2.20.134](https://github.com/OpenHFT/OpenHFT/releases/chronicle-bom-2.20.134)
